File: core/src/main/java/forpleuvoir/dhwu1a/core/message/messagesender/TempMessageSender.java
package forpleuvoir.dhwu1a.core.message.messagesender;

import com.google.gson.annotations.SerializedName;
import forpleuvoir.dhwu1a.core.message.base.MessageSender;
import forpleuvoir.dhwu1a.core.message.messagesender.data.GroupData;
import forpleuvoir.dhwu1a.core.user.Group;
import forpleuvoir.dhwu1a.core.user.Member;
import forpleuvoir.dhwu1a.core.user.base.Permission;

import javax.annotation.Nullable;

import static forpleuvoir.dhwu1a.core.websocket.base.ApiKey.*;

/**
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.message.messagesender
 * <p>#class_name TempMessageSender
 * <p>#create_time 2021/7/1 21:59
 */
public class TempMessageSender extends MessageSender {

    @SerializedName(MEMBER_NAME)
    public String memberName;
    @SerializedName(SPECIAL_TITLE)
    public String specialTitle;
    @SerializedName(PERMISSION)
    public Permission permission;
    @SerializedName(JOIN_TIMESTAMP)
    public Integer joinTimestamp;
    @SerializedName(LAST_SPEAK_TIMESTAMP)
    public Integer lastSpeakTimestamp;
    @SerializedName(MUTE_TIME_REMAINING)
    public Integer muteTimeRemaining;
    @SerializedName(GROUP)
    private GroupData group;

    protected TempMessageSender(Long id) {
        super(id);
    }

    @Nullable
    @Override
    public Member getUser() {
        if (bot == null) return null;
        Group group = bot.getGroup(this.group.id);
        if (group == null) return null;
        return group.getMember(this.id);
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/message/messagesender/TempMessageSender.kt
package forpleuvoir.dhwu1a.core.message.messagesender

import com.google.gson.annotations.SerializedName
import forpleuvoir.dhwu1a.core.common.*
import forpleuvoir.dhwu1a.core.common.data.GroupData
import forpleuvoir.dhwu1a.core.message.base.MessageSender
import forpleuvoir.dhwu1a.core.user.Member
import forpleuvoir.dhwu1a.core.user.base.Permission

/**
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.message.messagesender
 *
 * #class_name TempMessageSender
 *
 * #create_time 2021/7/1 21:59
 */
open class TempMessageSender protected constructor(id: Long) : MessageSender(id) {
    @SerializedName(MEMBER_NAME)
    var memberName: String? = null

    @SerializedName(SPECIAL_TITLE)
    var specialTitle: String? = null

    @SerializedName(PERMISSION)
    var permission: Permission? = null

    @SerializedName(JOIN_TIMESTAMP)
    var joinTimestamp: Int? = null

    @SerializedName(LAST_SPEAK_TIMESTAMP)
    var lastSpeakTimestamp: Int? = null

    @SerializedName(MUTE_TIME_REMAINING)
    var muteTimeRemaining: Int? = null

    @JvmField
    @SerializedName(GROUP)
    var group: GroupData? = null
    override val user: Member?
        get() = if (group?.user != null) group!!.user.getMember(this.id) else null
}